Dialog/CreditEditor_TableView.cpp

```diff
@@ -0,0 +1,237 @@
+
+     QModelIndexList index_list=this->selectionModel()->selectedIndexes();
+     for (int i=0;i<index_list.size();i++) {
+         QModelIndex model_index=index_list.at(i);
+         int column=model_index.column();
+
+         //If we are in the editable zone
+         if (column < 30) {
+             if (action_text.compare("One Tile") == 0) {
+                 this->model()->setData(model_index,QColor("#8080ff"), Qt::BackgroundRole);
+             } else if (action_text.compare("Upper Tile") == 0) {
+                 this->model()->setData(model_index,QColor("#ff8080"), Qt::BackgroundRole);
+             } else if (action_text.compare("Lower Tile") == 0) {
+                 this->model()->setData(model_index,QColor("#80ff80"), Qt::BackgroundRole);
+             } else if (action_text.compare("Delete") == 0) {
+                deleteFunction(model_index);
+             }
+         }
+     }
+
+}
+
+/// <summary>
+/// Triggered when a key is pressed, makes the delete key remove selected cells or
+/// insert letters or use directionnal keys to move
+/// </summary>
+/// <param name="event">
+/// An event used to know the key value
+/// </param>
+void CreditEditor_TableView::keyPressEvent(QKeyEvent *event) {
+    int keycode=event->key();
+    if (keycode == Qt::Key_Delete) {
+        QModelIndexList index_list=this->selectionModel()->selectedIndexes();
+        for (int i=0;i<index_list.size();i++) {
+            QModelIndex model_index=index_list.at(i);
+            int column=model_index.column();
+
+            //If we are in the editable zone
+            if (column < 30) {
+                deleteFunction(model_index);
+            }
+        }
+    } else if (keycode >= Qt::Key_A && keycode <=Qt::Key_Z) {
+        QModelIndexList index_list=this->selectionModel()->selectedIndexes();
+        for (int i=0;i<index_list.size();i++) {
+            QModelIndex model_index=index_list.at(i);
+            int column=model_index.column();
+
+            //If we are in the editable zone
+            if (column < 30) {
+
+                //Getting the background string
+                QString role=model_index.data(Qt::BackgroundRole).value<QColor>().name();
+                std::string role_string = role.toUtf8().constData();
+
+                //Write the letter in the cell
+                this->model()->setData(model_index,event->text().toUpper(), Qt::DisplayRole);
+
+                //If there is no backgroud, we take the one tile layout by default
+                if (role_string == "#ffffff" || role_string == "#3b3c3d" || role_string == "#000000") {
+                    this->model()->setData(model_index,QColor("#8080ff"), Qt::BackgroundRole);
+                }
+
+                //Selecting next index
+                if (index_list.size() == 1) {
+                    QModelIndex model_index=index_list.at(0);
+                    int column=model_index.column();
+                    int row=model_index.row();
+                    column++;
```

Comment:
If column is equal to 29 (as in, you are in the last column and type A - Z), this will set column to 30. Will the model then be in an invalid state?